### ISSUE TITLE
Errors appears in the App Builder (import): Build Status must be specified. Parse As Schema must be specified. File is not a valid Oracle APEX application export file.

### ISSUE THREAD

**Issue by fisal77** [jariolaine/apex-blog#10]
In the App Builder when I send files (blog-administration-latest.sql) & (blog-public-pages-latest.sql) to import the below errors appears:

3 errors have occurred
Build Status must be specified.
Parse As Schema must be specified.
File is not a valid Oracle APEX application export file.

Please help me.

<img width="950" alt="image" src="https://github.com/jariolaine/apex-blog/assets/23268583/2314b83d-4822-4a38-b97d-876540ce656f">

**Comment by jariolaine** [comment]
I'm not yet able to find what causes this because import works when running e.g. from SQLcl.

Workaround is use e.g. SQLcl to install applications.
Here is example how import applications:

```
set serveroutput on size unlimited;

-- Import blog admin application
begin
  apex_application_install.generate_offset;
  apex_application_install.set_auto_install_sup_obj( p_auto_install_sup_obj => true );
  -- Set your workspace name
  apex_application_install.set_workspace( '<WORKSPACE_NAME>' );
  -- Set your schema name
  apex_application_install.set_schema( '<SCHEMA_NAME>' );
  -- Set application id
  apex_application_install.set_application_id( 402 );
end;
/
@blog-administration-latest.sql;

-- Import blog public application
begin
  apex_application_install.generate_offset;
  apex_application_install.set_auto_install_sup_obj( p_auto_install_sup_obj => true );
  -- Set your workspace name
  apex_application_install.set_workspace( '<WORKSPACE_NAME>' );
  -- Set your schema name
  apex_application_install.set_schema( '<SCHEMA_NAME>' );
  -- Set application id
  apex_application_install.set_application_id( 401 );
end;
/
@blog-public-pages-latest.sql;

-- Query errors
select *
from user_errors
;
```

**Comment by jariolaine** [comment]
Have you connected right database and user? Errors indicate that database where you have connected doesn't have APEX installed.
Also check that user you use is assigned to APEX workspace where you going to import applications.

**Comment by fisal77** [comment]
Hi @jariolaine,
Thank you for your reply.
I follows your instructions as above but appears this error:

Error report -
ORA-06550: line 2, column 3:
PLS-00201: identifier 'APEX_APPLICATION_INSTALL.GENERATE_OFFSET' must be declared
ORA-06550: line 2, column 3:
PL/SQL: Statement ignored
ORA-06550: line 3, column 3:
PLS-00201: identifier 'APEX_APPLICATION_INSTALL.SET_AUTO_INSTALL_SUP_OBJ' must be declared
ORA-06550: line 3, column 3:
PL/SQL: Statement ignored
ORA-06550: line 5, column 3:
PLS-00201: identifier 'APEX_APPLICATION_INSTALL.SET_WORKSPACE' must be declared
ORA-06550: line 5, column 3:
PL/SQL: Statement ignored
ORA-06550: line 7, column 3:
PLS-00201: identifier 'APEX_APPLICATION_INSTALL.SET_SCHEMA' must be declared
ORA-06550: line 7, column 3:
PL/SQL: Statement ignored
ORA-06550: line 9, column 3:
PLS-00201: identifier 'APEX_APPLICATION_INSTALL.SET_APPLICATION_ID' must be declaredORA-06550: line 9, column 3:
PL/SQL: Statement ignored
06550. 00000 -  "line %s, column %s:\n%s"
*Cause:    Usually a PL/SQL compilation error.
*Action:   Refer to the following PL/SQL messages for more
           information about the error.

More Details :
https://docs.oracle.com/error-help/db/ora-06550/
https://docs.oracle.com/error-help/db/pls-00201/
--application/set_environment

**Comment by jariolaine** [comment]
It seems splitted export import works.
You can download whole repository.
Go to repository home page and click green "Code" button. Click "Download ZIP".
![image](https://github.com/jariolaine/apex-blog/assets/15383136/4c3fd381-fc83-4f46-bfa9-783788ddf278)

Extract ZIP and then go extracted folder -> APEX.
Zip folder blog_administration and create another Zip from folder blog_public_pages.
You can now import those created Zip files using APEX builder. Remember import first Zip containing blog_administration,

**Comment by jariolaine** [comment]
I managed identify problem. It seems when including Copyright Banner application export is corrupted. This is APEX bug.

**Comment by jariolaine** [comment]
Issue is fixed in [latest version](https://github.com/jariolaine/apex-blog/releases/latest/download/apex-blog-latest.zip).

**Comment by fisal77** [comment]
> It seems splitted export import works. You can download whole repository. Go to repository home page and click green "Code" button. Click "Download ZIP". ![image](https://user-images.githubusercontent.com/15383136/280518331-4c3fd381-fc83-4f46-bfa9-783788ddf278.png)
> 
> Extract ZIP and then go extracted folder -> APEX. Zip folder blog_administration and create another Zip from folder blog_public_pages. You can now import those created Zip files using APEX builder. Remember import first Zip containing blog_administration,

Thank you for your help. I've follow the above instructions but this error appears:

`Application parsing schema must have role CTXAPP or execute privilege to package CTX_DDL.`

![image](https://github.com/jariolaine/apex-blog/assets/23268583/8ac292e6-44c1-4aba-b365-d69e849dfa39)


Could you please help how I can give Application parsing schema role CTXAPP ?... I'm still new in Oracle DBA :( ....

**Comment by jariolaine** [comment]
Connect to DB as SYS and grant CTXAPP role to schema associated to your APEX workspace.

`grant ctxapp to <schema name>;`

**Comment by fisal77** [comment]
Thank you. my 'grant CTXAPP' problem solved. but appears new issue in blog_administration only & it's running as below screenshot:

<img width="715" alt="image" src="https://github.com/jariolaine/apex-blog/assets/23268583/afb61877-dc03-4585-b2e0-08e1a25ea20f">

Note: blog_public_pages was succussed in installation steps...

Please help me.

**Comment by jariolaine** [comment]
Unfortunately you are hitting another APEX bug 35410651 that is fixed in latest APEX 23.1 patch set bundle 35283657. You should apply that that patch before installing applications.
Or workaround is run supporting objects [database_objects.sql](https://raw.githubusercontent.com/jariolaine/apex-blog/master/APEX/Supporting_Objects/admin/database_objects.sql) scripts using e.g. SQLcl when connected to application parsing schema.
